Automattic/woocommerce-payments
# Custom email receipt for terminal card reader

Fixes #3558 

#### Changes proposed in this Pull Request
This PR adds a new custom email that will be delivered to a customer when they pay for an order with a card reader. 
Using the `woocommerce_email_classes` and `woocommerce_email_actions` hooks implemented in `WC_Emails` class we are able to leverage the email capabilities of the WooCommerce plugin.

This email will be triggered by the `payment_intent.suceeded` webhook that is forwarded from the WC Pay Server, whenever the payment intent is marked as sucessful.

The layout of the new email is based on the design of the emails that the customers currently receive when a new order is created.  You can find below screenshots of the html and plain text versions:








**HTML version**

<img width="600" alt="Screen Shot 2022-03-30 at 16 45 55" src="https://user-images.githubusercontent.com/1553182/160863088-1cd9994c-b6cf-4353-bc2c-53526f024bb6.png">



**Plain Text version**

```
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Your receipt for order: #59
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

Hi John,

This is the receipt for your order #59:

==========

WooCommerce Payments Dev

==========

60 29th Street
#343
San Francisco California 94110 US
+16267792059 help@mgasca-dev.jurassic.tube
2022-03-23 14:55PM

----------------------------------------

[ORDER #59] (MARCH 21, 2022)

Beanie with Logo X 1 = $18.00


==========

Subtotal:	 $18.00
Shipping:	 Local pickup
Payment method:	 WooCommerce In-Person Payments
Total:	 $18.00

----------------------------------------

Payment Method:	Visa - 0119
Application Name:	
AID:	
Account Type:	Credit


----------------------------------------

Thanks for using mgasca-dev.jurassic.tube!

----------------------------------------

WooCommerce Payments Dev -- Built with WooCommerce
```

#### Managing email settings

The new email settings can be managed from the `WP Admin > WooCommerce > Settings > Emails`:

<img width="1300" alt="Screen Shot 2022-03-25 at 19 44 25" src="https://user-images.githubusercontent.com/1553182/160183327-1e18cf89-d724-474c-9f37-ea38d98415a1.png">

From the details view the merchant can manage settings like the email type, subject, heading, ...

<img width="1312" alt="Screen Shot 2022-03-25 at 19 51 34" src="https://user-images.githubusercontent.com/1553182/160183624-3b0cccf0-ced7-4bee-9737-bdc53971db5f.png">
 

#### Testing instructions
*Prerequisites*
Your local setup is not likely to be configured to send emails. There are plugins that let you see the sent emails, like [WP Mail Logging](https://wordpress.org/plugins/wp-mail-logging/)

- Connect your local WC Pay client and server.
- Create an order with Cash on Delivery as payment method. Please use the proxied url of your local site.
- Start listening for webhooks in your local WC Pay server. You'll need this to trigger the email.
- Use the WooCommerce Mobile App to pay the order. You can follow this great guide to set it up locally: pdjVC1-9M-p2
- After the mobile apps process the payment correctly, go to `WP Admin > Tools > WP Mail Log`
- Check that there is a new entry in the emails table. Clicking on it you can access a preview of the email






-------------------


**Post merge**
